File: frontend/src/types/architecture.ts
/**
 * Types for the Architecture API v2 endpoints (ADR 0023) — mirrors
 * backend/app/schemas/architecture.py and the additions to
 * backend/app/schemas/indexing.py.
 */

export interface ArchitectureRepositorySummary {
  repository_id: string;
  name: string;
  full_name: string;
  domain: string | null;
  indexing_status: "pending" | "running" | "completed" | "failed" | null;
  last_indexed_at: string | null;
  node_count: number;
  node_counts_by_label: Record<string, number>;
  is_stale: boolean;
}

export interface DomainSummary {
  /** `null` groups every repository with no domain assigned. */
  domain: string | null;
  repository_count: number;
  node_count: number;
}

export interface ArchitectureSummary {
  total_repositories: number;
  total_nodes: number;
  total_cross_repository_edges: number;
  repositories: ArchitectureRepositorySummary[];
  domains: DomainSummary[];
  unindexed_count: number;
  stale_count: number;
}

export interface NodeTypeCounts {
  counts: Record<string, number>;
}

File: backend/app/schemas/architecture.py
"""Response schemas for GET /architecture/summary (ADR 0023) — the
org-scale replacement for ArchitecturePage.tsx's per-repository
GET /repositories/{id}/index fan-out.
"""

import uuid
from datetime import datetime

from pydantic import BaseModel


class RepositorySummary(BaseModel):
    repository_id: uuid.UUID
    name: str
    full_name: str
    # ADR 0023 — manual grouping only; None means ungrouped.
    domain: str | None = None
    # None = never indexed. Otherwise the latest IndexingJob's status
    # ("pending" | "running" | "completed" | "failed").
    indexing_status: str | None = None
    last_indexed_at: datetime | None = None
    node_count: int = 0
    # Excludes the base "GraphNode" label — every node carries it, so
    # it's never a meaningful per-type breakdown entry.
    node_counts_by_label: dict[str, int] = {}
    # True when `last_indexed_at` is older than the summary's own
    # staleness threshold, or when it's never been indexed at all — see
    # `architecture.py`'s `_STALE_THRESHOLD_DAYS`.
    is_stale: bool = False


class DomainSummary(BaseModel):
    # None groups every repository with no domain assigned — surfaced as
    # its own entry ("Ungrouped" is a frontend label choice), not omitted.
    domain: str | None
    repository_count: int
    node_count: int


class ArchitectureSummaryResponse(BaseModel):
    total_repositories: int
    total_nodes: int
    total_cross_repository_edges: int
    repositories: list[RepositorySummary]
    domains: list[DomainSummary]
    unindexed_count: int
    stale_count: int
